cdap-docs/examples-manual/source/examples/wikipedia-data-pipeline.rst

```diff
@@ -0,0 +1,140 @@
+A Cask Data Application Platform (CDAP) example demonstrating a typical data processing pipeline using CDAP Workflows.
+
+Overview
+========
+
+This example demonstrates a CDAP application performing analysis on Wikipedia data using MapReduce and Spark programs
+running within a CDAP Workflow: *WikipediaPipelineWorkflow*.
+
+This example can be run in both online and offline modes.
+- In the online mode, the custom action *DownloadWikiDataAction* reads the *pageTitleStream* is one output element
+of the `Facebook "likes" API <https://developers.facebook.com/docs/graph-api/reference/v2.4/object/likes>`__.
+For each event, it tries to download Wikipedia data for the page using the
+`MediaWiki Wikipedia API <https://www.mediawiki.org/wiki/API:Main_page>`__. It stores the downloaded data in the
+KeyValueTable dataset *wikiData*.
+- In the offline mode, it expects Wikipedia data formatted per the output of the MediaWiki API in the stream *wikiStream*.
+
+The MapReduce program *wikiDataToDataset* consumes this stream and stores it in the same KeyValueTable dataset
+ *wikiData*. Data can be uploaded to the *wikiStream* using the CDAP CLI.
+
+Once raw Wikipedia data is available is available from using either the online or offline modes, the
+*WikipediaPipelineWorkflow* runs a MapReduce program *WikiContentValidatorAndNormalizer* that filters bad records from
+the raw data, as well as normalizes it by converting the wikitext-formatted data to plain text. It then stores the
+output in another KeyValueTable dataset *normalized*.
+
+The *WikipediaPipelineWorkflow* then contains a fork, with two branches. One branch runs the Apache Spark program *SparkWikipediaAnalyzer*.
+This program consumes normalized data and runs topic modeling on it using the Latent Dirichlet Allocation (LDA) algorithm. It stores its
+output in the CDAP Table dataset *lda*, with one row for each iteration, and a column per topic containing the score for that topic.
+The other branch contains a MapReduce program *TopNMapReduce* that consumes the normalized data and produces the top N words in the dataset *topn*.
+
+Let's look at some of these components, and then run the application and see the results.
+
+The WikipediaPipeline Application
+---------------------------------
+
+As in the other `examples <index.html>`__, the components
+of the application are tied together by the class ``WikipediaPipelineApp``:
+
+.. literalinclude:: /../../../cdap-examples/WikipediaPipeline/src/main/java/co/cask/cdap/examples/wikipedia/WikipediaPipelineApp.java
+   :language: java
+   :lines: 24-55
+
+
+Running the Example
+===================
+
+.. highlight:: console
+
+Injecting data
+--------------
+The *pageTitleStream* consumes events in the format returned by the Facebook "Likes" Graph API.
+
+Once the application is deployed:
+
+- Inject a file of "likes" data to the stream *pageTitleStream* by running this command from the Standalone CDAP SDK
+directory, using the CDAP Command Line Interface::

```

Comment:
Like previous lines, this will not format correctly unless you add two leading spaces.